lib/Backend/arm64/LegalizeMD.cpp

```diff
@@ -373,7 +373,8 @@ void LegalizeMD::LegalizeIndirOffset(IR::Instr * instr, IR::IndirOpnd * indirOpn
                 indexOpnd = newIndexOpnd;
             }
 
-            Lowerer::HoistIndirIndexOpndAsAdd(instr, indirOpnd, baseOpnd, indexOpnd, fPostRegAlloc ? SCRATCH_REG : RegNOREG);
+            IR::Instr * instrAdd = Lowerer::HoistIndirIndexOpndAsAdd(instr, indirOpnd, baseOpnd, indexOpnd, fPostRegAlloc ? SCRATCH_REG : RegNOREG);
```

Comment:
nit: you could, alternatively, pass fPostRegAlloc to HoistIndirIndexOpndAsAdd, which would get rid of both the ternary at the end of both calls, and let you do the LegalizeInstr at just one spot.

Reply:
I considered that, but I thought it'd be good to keep Lowerer agnostic of that, and as a follow up I want to look into removing the need to pass that parameter around entirely, func already has a postRegAlloc flag that can probably be used instead.